### ISSUE TITLE
General project discussion

### ISSUE THREAD

**Issue by Heliomance** [Heliomance/HeroForge-Anew#53]
GitHub doesn't seem to have an easy way to have general discussion about the project, so I figured opening an issue made a reasonable workaround. Here - general discussion thread for all.

**Comment by TwistedFaith84** [comment]
Has every1 stopped checking/updating Heroforge for now? I've been noticing that no1 posted anything on or about Heroforge for atleast a good Month?

**Comment by Heliomance** [comment]
Yeah, I was wondering about that. It might be because I've slowed down a lot on releases, but I've added a dev branch with what I'm working on at the moment, and that didn't garner any attention at all.

**Comment by timoc** [comment]
FWIW, I work on this usually after my game, and its been on pause for a month. I will get back into it in the next week or so.

**Comment by TwistedFaith84** [comment]
So how is everyone doing? I know there hasn't been a lot of activity on here for a while so just thought I ask that and also see how things are coming along with the next update? Also do you have @Heliomance do you know about how long for the next update to be ready? Not trying to rush or anything just curious.

**Comment by torrasque666** [comment]
I would be more than willing to post a version with a new book or two. I'd just need to know which one people would prefer. I could do something as simple as adding a new book, like Drow of the Underdark. Or I could also see about doing something more complex, like implementing the Weapons of Legacy system. I'd just need to know what people would prefer.

**Comment by johnjstanton** [comment]
My group has been playing 5e for a while now. Does anyone have a favorite
(excel) sheet for that?

Thanks,
John

**Comment by TwistedFaith84** [comment]
@torrasque666 Drow Of The Underdark would be a nice addition. Weapons Of Legacy would be ok for later as it's mostly weapons and most of the feats in it go with said weapons and only 3 or 4 general feats IIRC.

@johnjstanton A nice excel character sheet for 5th that I use is ForgedAnvil's 5th Character Creater at you can get from ENWorld that he updates all the time. The link it is http://www.enworld.org/forum/rpgdownloads.php?do=download&downloadid=1234

**Comment by Heliomance** [comment]
Drow of the Underdark is probably the next book I'll implement. As for the next release, I'm afraid I can't give an estimate for that. I'm quite busy at work at the moment (I was previously working on HF on slow days in the office), and the next release is v8.0. It's a complete overhaul of the sheet structure, and there's a lot of stuff to do. It's going to be a while, sorry.

**Comment by TwistedFaith84** [comment]
That sounds great for Drow of the Underdark. It would be a great addition. As for the next release that's completely understandable. Personal and Professional life should always come first (in the order) and with what you doing with HF that's a lot of work. Friday, 23 October 2015, 03:00AM -05:00 from Heliomance < notifications@github.com> :

**Comment by TwistedFaith84** [comment]
## @torrasque666 out of curiosity if you uploaded your new version with those 1 or 2 books does that version include the easy fix I found for the feats that requires specific deities. 

Sent from myMail app for Android Thursday, 22 October 2015, 07:34PM -05:00 from torrasque666 < notifications@github.com> :

**Comment by TwistedFaith84** [comment]
@Heliomance please correct me if I'm wrong but IIRC didn't you say that you have a better handle on javascript then VBA? Or was that someone else in here. I do have a reason for asking but I'll get to that later. Friday, 23 October 2015, 03:00AM -05:00 from Heliomance < notifications@github.com> :

**Comment by Heliomance** [comment]
No, I don't know javascript at all. Python and C# are my languages of choice. I've picked up basic competence in VBA from poking at HF though.

**Comment by TwistedFaith84** [comment]
Ahh ok. well what I was think was if someone knows java and was willing to recode it might be better to turn HF into a google spreadsheet (if it would work. I recently read something about GoogleAppScript or something like that). I know that that would be a ton of data to transfer over but it would be a great place to store it and if i'm thinking correctly it would make contribute to the data adding alot simpler and everything would be in one place. Just an idea though.

**Comment by torrasque666** [comment]
i'd prefer not to work on DotU until I know the new sheet structure. Don't want to waste all my hard work. JK. But seriously, I'd rather know the new structure before working on adding a new source. Hell, I'll even work on the new format. Overseen by @Heliomance of course.

**Comment by TwistedFaith84** [comment]
I have a question and I only ask cause the person that asked about the weapon data kinda made me think. Is it possible on the Attack Sheet when selecting a weapon you can type what weapon you want (that is listed) and have it pull the info that way? I know there is a way to do it as it is done on a few other sheets. It would make selecting attacks faster (ie if you want a longsword you can start typing longsword and as you type the list narrows down to longsword).

**Comment by torrasque666** [comment]
@TwistedFaith84 sounds like..... I wanna say Combo Box maybe?

**Comment by Heliomance** [comment]
It's possible, it means changing the drop-downs to work off cell data validation instead of form controls. I can look into it, but it does change how the data is stored (form control drop-down stores it as a number, the position of the selected item in the list, whereas using data validation stores the actual string) which means a fair few other formulas would have to be changed.

The other thing I could potentially do to speed up looking for weapons is add a checkbox to restrict the list to weapons you're proficient with.

**Comment by TwistedFaith84** [comment]
@Heliomance Na that is OK. It's fine the way it is for me I just wasn't sure how much would actually have to be changed to do something like that.

@torrasque666 I'm not really sure. It might be.

**Comment by Heliomance** [comment]
Ye gods but familiar skills are a mess. I'm trying to sort them out at the moment and I have no idea what's going on!

...damn. I'm going to need to store default ranks for creatures, I was hoping to drop them. Right, time to come up with a new format for holding skills!

**Comment by robertkety** [comment]
Extra skill column that was not completed properly.  Creature Info tab,
Column AM, Iaikutsu Focus.  Eliminate that column and all will be well.

**Comment by Heliomance** [comment]
No it won't. I'm amalgamating the RaceInfo and CreatureInfo tables, which store skills differently to each other. I need to come up with a format that works for both of them. I'm also intending on pulling MonsterForge in to work off the same data sheet as well, so at some point I'll need to inspect that to see what info it expects.

I was going to just use the RaceInfo format for all skills, as it only takes up one cell and has a helper function already written for it. Also changing that to the CreatureInfo format would require restructuring large parts of the Skills tab which is probably the most impenetrable part of the whole sheet. Trouble is, that doesn't store default skill ranks, which things like familiars and animal companions - not to mention MonsterForge - will need.

**Comment by TwistedFaith84** [comment]
Well due to a messed up install/upgrade my Microsoft Office is no longer working. I tried just about everything (I haven't reformatted my drive yet) and it refuses to work. So after I get my drive backed up (it'll take me some time to see want I need to/want to keep) I'm stuck with no MS Office. So with that said any suggestions on which Office I should install after the format? 
My computer specs are 
AMD Athlon 64 X2 Dual-Core 4400+ 2.30 GHz
2 GB DDR2 Ram (only have 2 slots and cant really upgrade with out upgrading the CPU)
No Videocard

I know its a sad computer but I can only work with what I have and with no cash coming in I cant buy better.

**Comment by Heliomance** [comment]
v8.0 is transitioning to .xlsm format, so you'll need Office 2007 or above.

**Comment by TwistedFaith84** [comment]
Ok cause I was think of going back to Office 2010 as that still gave me the smoothest experience. Everything above 2010 to me just felt like it had alot of LAG. Granted I'm sure if I had a better computer in general it would have had (or felt like) less LAG.

**Comment by TwistedFaith84** [comment]
So Github Has a slightly new style. Has anyone else switched over? And if so how do you like it?

**Comment by Heliomance** [comment]
Hah! Yes! Got familiar skills working! Now to copy the fix over to the Animal Companion sheet, then there's just a whole load of data entry to do to update the CreatureInfo sheet to the new format >_>

**Comment by Heliomance** [comment]
_twitch_

> Get animal companions almost finished
> Excel crashes
> Realise I haven't saved around two hours
> FFFFFFUUUUUUUUUUU

**Comment by TwistedFaith84** [comment]
Out Of Curiosity can the next update (8.1 or 8.0.1 how ever its going to work) be that instead of adding new books; can it be the completion of the book that need to be completed (ie I know that their are feats that still need to be added as well as Prestige Classes, Classes (mostly Alt. Classes), Magic Items, Spells(mostly Buffs/Debuffs added to the Buffs Sheet), and finally Equipment(Mostly Weapon and Armor Options (like materials, enhancements, and especially stuff like Dragonscale Husk))

**Comment by Heliomance** [comment]
Equipment needs a complete overhaul, the way it works at the moment just isn't good enough. The trouble with that is that I have no idea what to do with it. Once I'm done with 8.0 I may open a discussion thread about it.

**Comment by TwistedFaith84** [comment]
That's understandable. The equipment DOES NEED a MAJOR overhaul. I know I suggested maybe something similar to YAPCG. But yea we can discuss that after 8.0 is finished and released. As for the the rest of the missing stuff it would be nice to have after 8.0 is done while thinking about how to Redo The Equipment.

**Comment by ThAlEdison** [comment]
So open questions then,
Should you be able to put more than one/each slot of equipment on the
character sheet.
If so, how should the choice be implemented? Drop down?
How should magical bonuses be handled?
Should equipment abilities be listed with class /racial /feat abilities? If
so, should it only list the abilities for worn equipment?
How should extra dimensional storage space be handled for weight?
Should there be a check for selecting melds vs equipment?
Should the epic extra slot feat be handled?

**Comment by TwistedFaith84** [comment]
Ok going in order of questions asked
1) Yes but there should be a checkbox or something for weapons and armor as to which one your currently using/equipped cause just because you have a +5 Ghost Touch Longsword of Defending in your stash doesn't mean your currently using it so you cant have the up +5 added to your AC. Same goes to Armor.
2) Yea drop down would work but I'm thinking more something like just typing the enhancements in would be a good way to do it as well and if you have more then 1 Enhancement you just separate them with a colon (;) or something that way if someone want to keep spending the gold to have 5+ enh. they could.
3) It could but maybe better having its own separate sheet although having the calc.work properly could be a bit of a challenge figuring out what you have equipped and using vs what you have in your stash.
4) Extra-Dimensional space handling is relatively easy for weight. There could be a sheet strictly for stuff like that where on that sheet you have like 5 or so different sections for your Extra-Dimensional Storage with a drop down for choosing what storage is being used for that section and also being able to use like 2 of the same item (ie 2 Bags of Holding Type IV) and still have it calculate them as different storage.
5) IDK as I've never dealt with using melds yet so someone who has and know the rules better for that can give their opinion.
6) Yes as well as items that are slotless (Ioun Stones) and if I recall there are items (not just feats) and/or abilities that give you an extra slot

**Comment by Heliomance** [comment]
Just released to dev, animal companions and familiars should now be working! Currently only Baboons, Badgers, Bats and Bears (Black, Brown and Polar) have the new skill format done for them, so those are the only critters that skills will work properly for. If anyone wants to do some data entry for me, the format for skills (in the CreatureInfo sheet in the data folder) is as follows:

`<skillname>:<ranks>+<racialbonus>`

Only skills listed in the statblock should be entered, any skill mentioned in the column will be treated as a class skill for that creature. If it has no ranks and no racial bonus, the colon and everything after are unnecessary. If it has no racial bonus, the + is unnecessary. If it has a racial bonus but no ranks, you do still need to enter zero for the ranks like so:

`Climb:0+8`

Skills should be separated by a comma, no spaces. Racial bonus should _not_ include size modifiers, ability modifiers, or anything of the sort.

**Comment by ThAlEdison** [comment]
I've been doing some data entry.
1.The Oriental Adventure animals (with the exception of the toads) don't
actually exist; they're a partial entry for the Hengeyokai race because
they lack mental stats, skills, hit dice, and feat selection.
2. How are we handling the non-bonus feats for animal companions/familiars?
3. Almost every animal has a +2/+2 skill feat (Alertness, Agile, Stealthy)
this affects the entries.
4. It seems like animals from the Player's Guide to Arcanis d20 forgot to
use the 4x multiplier for skill points at first level (or they universally
have 3 unspent skill points).

~Erick A. Cobb

**Comment by ThAlEdison** [comment]
What's the best way to get the file to you.  I've put in some of the
missing data from Ankheg through Clockwork Mender.  I've put in skills all
the way to Dire Lion.
Still unsure about what to do for the feats, I could throw them into the
bonus feats, but most aren't actually "bonus".
If the data is ever used for monster building, having those feats in the
bonus feats area won't make sense.

~Erick A. Cobb

**Comment by Heliomance** [comment]
Hrm, true. I am planning on using the same data sheet to resurrect MonsterForge at some point. I'll think about that later. 

Best way to get the sheet to me is with a pull request, which is half the reason I started spring out the data tables into separate files. Makes collaboration a hell of a lot easier.

**Comment by TwistedFaith84** [comment]
Is there a way to have the Aging Effects be Calculated? I have a idea on how it could be done if it wasn't in the works already. My idea is Add 3 Columns to the Race Info right after the Age Columns for the Min Age for each Age Bracket. Then just figure out a formula for the Calculations on whichever sheet that they need to be on. Finally have a Checkbox (with formula attached) on the Stats Sheet (next to the Age) for the exclusion of the Neg to the Physical Stats with a box next to it for how much of the neg to take off. Or something like that.

**Comment by Heliomance** [comment]
It would be doable, yeah. Low priority though, and age information isn't available for every race - or even most races - anyway.

**Comment by ThAlEdison** [comment]
Another issue is Abilty scores.  I think you're already working on this,
but the Familiars/Companions are using the full stats while other races
only have the racial modifiers.  I've got maybe a temp workaround using the
Adj columns, but I'm not sure that's the best way to do things.

~Erick A. Cobb

**Comment by Heliomance** [comment]
When I'm done, everything's going to run off full starts, thanks for reminding me. There's also going to need to be an override column for races where the stat adjustments given in the "as characters" section don't match the base stats (see aasimar for an example IIRC), which is annoying.

**Comment by ThAlEdison** [comment]
Aasimar use the non-standard array 13,12,11,10,9,8 instead of
11,11,11,10,10,10. But there's no way to directly determine this in most
cases.
The other thing I was thinking about today is dragon age categories.  It's
easy for familiars (a dragon older than wyrmling is ineligible to be a
familiar).  But for special mounts and monsters it becomes another table
with about 80% of the columns of the main table.  While there are some
patterns, there's also some odd breaks in the pattern.

~Erick A. Cobb

**Comment by TwistedFaith84** [comment]
Is there a way to have Spell Points Be calculated for those that want to use Spell Points? Also can the Conditional Modifiers Box finally be fixed to show the Cond. Mods. like it was intend to and once did? if not can it be removed to make room for other stuff that can go their in its place (IE you can move SR, ASF, Act. Points, And Sp. Points to that location and have room for maybe 2 more attacks or something. I'd rather have the Cond. Mod Box fixed but not sure what other ppl would want.

**Comment by TwistedFaith84** [comment]
Hey just thought of something wouldn't it be better to get rid of the Bonus Feats Class selector altogether and just have the bonus feats show up in blue color when you select 1 or more classes that grant bonus feats? I mean it would be alot easier instead of selecting 1 class then switching to another in the selection just to select you bonus feats for all classes. The only prob that I can see though is when selection multiple classes that grant the same bonus feats or grant bonus feats to feats you get for free but as far as the free feats go if the free feats override the bonus feats automatically with the teal color (i guess it's teal) instead of the blue then you know that those certain feats are already free so you dont need to select them as a bonus.

**Comment by torrasque666** [comment]
Hmm... Now I'm wondering how much processor power it would take to pull changes from another sheet, since its moving to a "each data sheet is on its own now" format. Also wondering if a closed excel sheet can still get updated results. Like I have a "Feats" sheet that pulls from a separate "Feats Data" which would need to be updated with any and all changes like the current feats sheet does.

**Comment by TwistedFaith84** [comment]
@torrasque666 thats an interesting question. Guess the only way to find out is trial and error with low end PC's and also different versions of excel (ie everything before 2013 doesn't hog the RAM and/or CPU as much as everything 2013 and after).

**Comment by torrasque666** [comment]
Just tried. Can't maintain references to a closed sheet. As such, experiment is a failure. Must try to implement YAPCG-esque feat selection in other means.

**Comment by Heliomance** [comment]
If you guys have any ideas on what to do with the Feats tab, could you head over to issue #68?

**Comment by TwistedFaith84** [comment]
@torrasque666 what if you had those closed Feats Sheets Merged into 1 Workbook thats is specifically for the feats? or will HeroForge not work with closed Sheets/Workbooks?
@Heliomance I know having the MaxHP work again just requires the re-linking of the checkbox to the proper cell but is it possible to have it work for Animal Companions and Familiars as well?

**Comment by TwistedFaith84** [comment]
Hey I know its still a little early but I wanted to say Happy Holidays To Everyone. I wish you all a _**Joyous, Peaceful, Exciting, Loving, and Wonderful**_ Holiday. May The New Year bring you all _**Peace, Love, and Happiness**_.

**Comment by TwistedFaith84** [comment]
So How is everyone doing???

**Comment by ThAlEdison** [comment]
I've been busy, I should take time tomorrow to submit the work I've done.

**Comment by TwistedFaith84** [comment]
@ThAlEdison Ahh so what have you been working on? Still trying to do something with the feats right now or something different?

**Comment by ThAlEdison** [comment]
I've been doing some data entry on animals, which may undergo a change at
some point, since there isn't really a way right now to track animal feats.

**Comment by Heliomance** [comment]
Feat storage may change at aome point, the rest shouldn't. I think the current format should work fine, so your efforts shouldn't be wasted!

**Comment by TwistedFaith84** [comment]
Sounds good

**Comment by Heliomance** [comment]
I've just finished pulling all the formulae out of the WeaponInfo sheet, making it a purely static data table. That meant I could convert it to a .csv, which not only made it smaller, but also means that multiple people can submit changes to it without interfering, because GitHub can now diff it properly! Released to dev, if anyone wants to look. Note that natural weapons currently won't work, I need to work out how I'm going to do them.

**Comment by Heliomance** [comment]
Getting natural attacks working again was easier than I expected. Dev version should now be fully functional in terms of attacks, running purely off the external sheet. Let me know if there are any issues!

**Comment by ThAlEdison** [comment]
I just now deleted my creatureinfo by accident, and my backups are
corrupt.  Will try to redo.  But feel free to make any changes you want
first.

~Erick A. Cobb

**Comment by ThAlEdison** [comment]
I'll probably redo it with less ambition, but I likely won't start again
until the middle of this month, so if there was anything you were holding
off on, please go ahead.

~Erick A. Cobb

**Comment by TwistedFaith84** [comment]
ok out of curiosity when downloading the dev version of HF should I just dl the xlsm file or the xlsm file and the data folder?

**Comment by ThAlEdison** [comment]
The xlsm file references the data files, so I'm pretty sure all of them are
necessary.

**Comment by Heliomance** [comment]
Data folder as well. That's where all the data tables are being moved to, an impressive amount of stuff will break without that.

**Comment by TwistedFaith84** [comment]
ok so does the data folder have to be in a "Data Folder" thats in the same folder as the xlsm? or do they all have to be in just the same folder as the xlsm?

I only as cause I know that if the files are not in the proper place then things will get broken until everything is in the proper place.

**Comment by Heliomance** [comment]
The structure they're in on github is the correct structure. Just download the whole archive.

**Comment by TwistedFaith84** [comment]
Ok. btw if making the WeaponsInfo Sheet could be made into a .csv file to compress the size why cant all the other Info Tables be converted that way. Is it cause of all the VBA code on those sheets or can they be converted to get a smaller total file size?

**Comment by Heliomance** [comment]
They are going to be, primarily for diff and merge purposes rather than filesize. .csvs cant have formulae in, though, and WeaponInfo is just the first one I finished stripping all the formulae out of.

**Comment by TwistedFaith84** [comment]
ahh ok.didn't know that about csv files

**Comment by Heliomance** [comment]
Actually to be honest, seeing as Excel apparently refuses to read from csv data files unless they're open (I've found a workaround, don't worry), it probably could cope with formulae in there (as if they're open to read from them it can calculate). Regardless, I want to try and make all the data sheets purely static, and converting to .csv is a handy way of indicating that I've finished pulling out all the formulae from the sheet.

**Comment by TwistedFaith84** [comment]
I just realized that what your trying to do with the data sheets is going to be very hard work; cause once you remove that all the formula from the data sheets to make them completely static .... well all that formula has to go somewhere to keep HF working properly if I'm thinking correctly. Which means either it's gonna have to be converted to VBA then added to HF or it's just has to be added to HF somewhere else.

**Comment by Heliomance** [comment]
There's not many formulae in the data tables, the ones that are heavily formula-based I'm leaving as they are. I'm only splitting out the ones that are mostly static anyway. The bits that do need calculating, most of it is on the new interfaces tab, with a few other bits and pieces where they seem appropriate.

**Comment by timoc** [comment]
The way i would do it (sorry guys don't have time to contribute!) is to write a simple VB macro that performs the cell calculation, replace it in one place, then bulk replace the expression in the XML.

**Comment by TwistedFaith84** [comment]
So how is everyone doing? How is Everything going? Any updates as to how much longer for v8.0 of HF (just asking not impatient or anything)?

**Comment by torrasque666** [comment]
Hey @TwistedFaith84 , do you still have all those other Forge Files? i tried hunting them down through the old commits but the link was dead.

**Comment by TwistedFaith84** [comment]
@torrasque666 Yes I Do. All the ones that I have are in my dropbox. Here's the link to all the DND Tools that I have including most (if not all) of the old HeroForges (look in NZComputer folder).
https://www.dropbox.com/sh/9n4crdfvrc10bz1/AADDbxznLSwiwxJgUO1bmVVma?dl=0

**Comment by TwistedFaith84** [comment]
Have people given up checking here for anything?

**Comment by Zukibird** [comment]
Nope

**Comment by Keldin** [comment]
> Have people given up checking here for anything?

I'm still here.

**Comment by TwistedFaith84** [comment]
I know I check here daily. To see if someone has another issue that I might be able to help with, If any of the data files have be updated, and to see if @Heliomance has upload v8 (mostly the first two cause I know v8 will take a while to get done every thing she wants to get done as well as work and life). But yeah I know for a while HeroForge-Anew was very active and then poof nothing so I was just curious.

**Comment by torrasque666** [comment]
I just wait to see if there's a message in my inbox regarding this. Does that count as "checking"?

**Comment by TheNarf** [comment]
Im subscribed to All threads and follow / stalk this project very closely
Den 03/04/2016 14.30 skrev "torrasque666" notifications@github.com:

**Comment by Heliomance** [comment]
I'm still around. There's deadlines coming up at work so much busyness at the moment, but once they're past it's possible I might have more spare time to work on this again. I haven't abandoned the project, promise - version 8 will be out eventually.

**Comment by TwistedFaith84** [comment]
Just out of curiosity is it at all possible to somehow reasonably add Illithid Savant PrC from Savage Species? I know that with its abilities it can gain all skills, class features, feats, and special attacks and/or special qualities just by eating every brain it can; essentially it can become everything with just about everything after gaining the 5th level of the PrC if it ate enough brains.

My guess is if it were to be added it would need it's own PrC Sheet to choose Class Features, Special Attacks and/or Special Qualities of everything available in HeroForge.

**Comment by PodPGH** [comment]
Just as an FYI, you can still get to the download files from nzcomputers... go directly to
http://nzcomputers.net/heroforge/downloads/
It's not pretty... but I pulled some files trying to find a solution to my "need a way to do awakened animal as character" issue. (using PCGen currently, which will fail with PrC) They didn't help completely, but I'll bring that up as a separate issue/bug here when I get a chance.

Thank you everyone for the hard work on this! I know that many people are using different systems by now, but the campaign I'm in is very much still 3.5 and doing anything without  HF would be painful, even if I'm still on version 6... Need to convert my primary character still. :)

**Comment by TwistedFaith84** [comment]
Hey guys I have a question. I'm looking for either a program, an Excel file, or even a webpage for Spells Generation that If I remember correctly was similar to SpellForge and would list your spells that you selected (or can cast) but if you had and metamagic feats would also list all your spells with any and all combination/s of metamagic feats you have @ their proper level (up to your current max spell level). If anyone knows what I'm talking about and can help me re-find it please do.

**Comment by Heliomance** [comment]
I actually did some work on HeroForge today! Work has calmed down a lot and I'm stalled out on another project I've been working on, so I think development here is going to start up again over the next few weeks. 

I didn't get as much done as I hoped, though. Excel crashing on you three times in a row has a way of rapidly destroying motivation >_>

**Comment by TwistedFaith84** [comment]
@Heliomance Yeah I know the feeling with excel crashing. Glad to hear that Development is going to start back up. Out of curiosity how would I add subtypes to characters. ie in a current game I'm playing a Human Rogue that happened to touch a few Pillars Of Creation and now I'm a Human with a Plant Subtype. Another party member touched different Pillars and is was turned into an Undead and another was turned into a Construct (I forget with kind at the moment).

**Comment by torrasque666** [comment]
Ok, so I've decided to shamelessly crib from YAPCG for spellforge, but I'm having a sliiiiight issue when it comes to spells. For limited-known-casters (Bard, Sorcerer, Assassin, etc) I could just make it so that they can only select up to the - wait a minute. You can cast spells out of higher slots but not select them using higher slots right? That simplifies things a bit I guess. Unless anyone else has any other ideas? [I'm uploading a new Spellforge version (WIP edition though.)](https://github.com/torrasque666/Spellforge-Anew) in case anyone wants to take a look at it.

**Comment by TwistedFaith84** [comment]
@Heliomance @timoc @torrasque666 @johnjstanton @robertkety @PodPGH @TheNarf @Keldin @Zukibird @ThAlEdison **_So How is everyone doing?_** Any news on the Dev side of **HF** and/or **SF**?

**Comment by timoc** [comment]
Nothing from me, I've not played 3.5 in a while so urgency has lessened.

**Comment by Heliomance** [comment]
Working on it again today. Going to try and get past the repeated-crash bug that's been destroying my motivation.

**Comment by Heliomance** [comment]
Just pushed a version to dev. Big change on this one - you need to open the Launcher file, not the main HeroForge file. Things should be mostly working, I think.

**Comment by torrasque666** [comment]
So quick question. What is the launcher file supposed to do? Because right now I'm trying to use it and its just sitting there. Not even a spreadsheet coming up. Unless there's some change I need to make to my Excel 2016.

**Comment by Heliomance** [comment]
At a guess, you may need to change your security settings so that it can run macros without asking? What the launcher is supposed to do is open the data csv files invisibly in the background, then open the main HeroForge file so that it can read from the data files.

**Comment by torrasque666** [comment]
Yup, that fixed it. 

```
  From: Heliomance <notifications@github.com>
```

 To: Heliomance/HeroForge-Anew HeroForge-Anew@noreply.github.com 
Cc: torrasque666 dzsenak@yahoo.com; Mention mention@noreply.github.com
 Sent: Wednesday, August 17, 2016 10:30 AM
 Subject: Re: [Heliomance/HeroForge-Anew] General project discussion (#53)

At a guess, you may need to change your security settings so that it can run macros without asking? What the launcher is supposed to do is open the data csv files invisibly in the background, then open the main HeroForge file so that it can read from the data files.—

**Comment by Heliomance** [comment]
There is WAY TOO MUCH calculation being done on the CreatureInfo sheet. Stripping it all out to convert it to a csv is being a touch nightmarish.

>_<

**Comment by Heliomance** [comment]
Dragons. I hate dragons. Not only are there innumerable different types, each of which comes in 12 varieties, but there are three different ways to Wild Shape into them, each with different restrictions.

Also, Wild Shape runs off Alternate Form, which grants (Ex) Special Attacks, but not Special Qualities. That's all  well and good until you realise that the Dragon MM entires don't actually spell out which of their abilities are Special Attacks and which are Special Qualities.

Why do I do this to myself?

**Comment by Heliomance** [comment]
I think I've now made lycanthropes work properly with the new datasheet format. If a few people could download the latest development version and have a play with them to see what inevitably breaks, I'd appreciate it.

**Comment by Heliomance** [comment]
And I've finally finished pulling all the calculations off the CreatureInfo sheet! It's now a csv, which means it's diffable, which means that multiple edits can be merged properly! If you were waiting until I'd finished making changes to CreatureInfo before starting to add more creatures to it, it's now safe to go ahead.

**Comment by TwistedFaith84** [comment]
So How is everyone? How is everything going? Anything New and Exciting? How is Heroforge Anew coming along? I noticed that there hasn't been any new updates in awhile? How much longer (estimated) till the release of v8.0?

**Comment by Heliomance** [comment]
I'm on leave from after tomorrow, I'll try and get some more work done over Christmas. Honestly, though, the complete lack of activity here's been something of an enthusiasm killer :/

**Comment by ThAlEdison** [comment]
I've never been that good on involvement, and I've had life stuff going on.
I'll likely pick up again when I start working on modules for the 3.5
campaign my group does at conventions. But that's not for a few months.

**Comment by TwistedFaith84** [comment]
@Heliomance Yeah I know and understand. but I think the reason for the lack of activity on here is cause everyone is waiting for v8.0. I know myself I do check here at least once a day but because of there not being any activity I feel the same way as you... it is an enthusiasm killer. I try and help with anyone that reports any problems (if I can actually help) but mostly I'm looking and waiting for v8.0 that way I can really start helping with data entry or looking for bugs and finding solutions for them.

@ThAlEdison Yeah I'm the same. I had Life stuff happening as well. Although I'm better at being involved but when life stuff started picking up I've become less involved.

Well till next time. See y'all later and just in case Merry Christmas and Happy New Years.

**Comment by Heliomance** [comment]
@TwistedFaith84 if you want to do data entry, the CreatureInfo sheet is ready to do that on. That's now in its final form, and isn't going to change in structure before 8.0. The fact that it's now in csv format means that GitHub can properly track changes in it, and merge edits from multiple users should that happen. If you do decide to add things here, it's probably best not to do anything with Dragons. I haven't quite decided how I want to implement them, there are so many variables to track it's a bit of a pain.

The WeaponInfo sheet is in the same state, should you want to do anything with that - adding sourcebook references to the weapons would be useful, as that'll let me add functionality to filter available weapons by source.

I may give up on splitting out ClassInfo to be honest - there's so much calculation going on in that sheet I'm not sure it would be practicable, and it wouldn't allow easy merging of added classes anyway unless I also split out the Class Abilities sheet, which would need a total overhaul to be useful. I'm also wavering on my idea to split out the Tables sheet - there's quite a lot of calculation going on there, and it's not neatly organised or anything, rather, it's a host of small tables that rarely need editing. Other people aren't actually going to need to modify that very often.

If I decide not to split out any more sheets this release, which is looking plausible, then I think I'm down to bugfixes before I have a stable 8.0 alpha to release. I want to move to XML-based saving before 8.0 proper, to improve forward-compatibility, and that's going to be a fairly large project, but also an interesting one. 

Custom races may or may not be fixed before 8.0, it depends how complicated it turns out to be.

**Comment by johnjstanton** [comment]
Your spreadsheet is D&D 3.5 only, correct? I have been playing only 5e for
about a year. We actually enjoy the new system much to my surprise: it is
simple but versatile.

**Comment by AkaSha1** [comment]
Hello !! 
i take a chance... i try to find a new character sheet that are working with excel 2016. i know that your group work on a new one  and want to know if the bugs are fixed already or if you have a spreadsheet working ? try to find it on this site or among your discussions but... still playing in 3.5, and those spredsheet are pretty useful !! thanks !!

**Comment by Heliomance** [comment]
@johnjstanton: yes, 3.5 only. 

@AkaSha1: not quite certain what you mean. The spreadsheet absolutely works, the latest stable version is on the Releases page.

**Comment by AkaSha1** [comment]
Ok ! i ask because i was thinking that the one on the release page was the one that we're not able to save, my error and the release date was last year so, i was also thinking that you'll made a new one since then ;)
My mistake, and i continue to read thread !!!
Thank you very very much for all you've done so far !!!

**Comment by TwistedFaith84** [comment]
@Heliomance Yeah I would love to do the Data entry but its gonna have to wait till I get back into town. I didn't want to be near any of this coldness or snow (I heard it was gonna be pretty bad this year where I'm from (Chicago)).

**Comment by TwistedFaith84** [comment]
Is HF still being worked on? Just curious as it looks like there is still no new release in sight anywhere in the near future.

**Comment by Heliomance** [comment]
I... haven't worked on it in a while, no. This page is still sitting in my constantly-open tabs, judging me, though - I'll get back to it at some point, I promise!

**Comment by TwistedFaith84** [comment]
LOL I've got it sitting in a constantly open tab as well. I hope sooner then later. I know everyone has lives (such as work, family, and/or friends) that they have to deal with.

**Comment by TwistedFaith84** [comment]
So here we are the LAST day of April.... How is everyone doing? Anyone get married/divorced? Have any kids or any other life changing events?

**Comment by TwistedFaith84** [comment]
How is v8.0 coming along? Any chance it gets released soonish?

**Comment by Heliomance** [comment]
I actually did some work on it the other day! I'll put up an alpha version of 8.0 with all the 8.0 tagged bugs fixed soon (hopefully within the week, work permitting), but I'm not going to call it 8.0 proper until I've got XML saving in. I've not started that yet, and it's probably going to take a while to make it work - it's quite possibly the biggest single change so far.

**Comment by Heliomance** [comment]
v8.0 alpha has been released!

**Comment by Okinawa3d** [comment]
Thanks much for all the work you put into making this effort happen.

VR
John Powers


MAKE READY!!!
COL Powers
Gulf Division, 75th Training Command: Building Army Reserve Readiness through Mission Command Training

JOHNNY J. POWERS
Colonel, FA57-Simulations Operations
Division Commander
Gulf Division/75th Training Command
 
1400 Golden Acorn Dr
Birmingham, Alabama 35244

Phone: 205.987.3903 (W)
Phone: 205.601.1610 (C) 
Email: Johnny.J.Powers2.mil@mail.mil




-----Original Message-----

**Comment by Kartemest** [comment]
Greetings from Italy :) Thank you for all the hard work you put into this project: it has helped me and a few other people from my group a lot. I'm sure many people from all around the world appreciate it.

**Comment by vikingeric9** [comment]
For someone who hasn't played since 2.0 and just coming back, this is a miracle of no small proportions. I wish this was still going

**Comment by Kartemest** [comment]
I think it is still going, just a bit slownly?

**Comment by JoeyD473** [comment]
I hope its still being worked on. I would love to see a few more books added,

**Comment by TwistedFaith84** [comment]
I'm pretty sure this isn't being worked on anymore. And if it is @Heliomance is just very very very busy with work and life. Unless she happened to get a C&D from Hasbro/WotC

**Comment by Heliomance** [comment]
No, I'm not actively working on this anymore I'm afraid. My old job, there were a few years where I was massively underworked, and I started work on this to have something to do during the day so I didn't go mad. Since then I've changed jobs a couple of times and no longer have that problem, and I don't play 3.5 anymore. I'm not saying I'll never come back to HeroForge, but I'm not intending to resume active development any time soon

**Comment by Zukibird** [comment]
Any chance you can post your latest sheet (if you still have it?)

**Comment by johannes-z** [comment]
There was the discussion once to move the project from excel to a standalone program, but I haven't read about that anymore in a long time. I've been using heroforge a lot, but Excel is really annoying (one of the main reasons why I don't use it anymore). Since the project became stale, I would like to know if somebody wants to work on that or rather help me working on that. I'm a programmer, but I don't want to do this alone. Also, I'm still unsure about the technologies. The *best* that comes to mind is some javascript and sqlite databases...

**Comment by Heliomance** [comment]
So coming here to answer that comment sparked a little enthusiasm in me for working on it again :P

As regards standalone, I was intending to first change the save format so it saved to JSON, which would mean that saves from the current version could theoretically be loaded into the standalone. I'm working on that at the moment, and by the Nine Hells VBA is an obnoxious language. Trying to make it work with JSON is a royal pain in the arse, and - now I'm actually coding professionally and have got to use actual proper IDEs - the inbuilt editor for it is so limited in its features it's painful.

I hadn't sketched out much on how the standalone was going to work, but I'd planned to do it in C#, that being what I know best.

**Comment by vikingeric9** [comment]
if theres a donate page, I'd be happy to donate some (what i can) if someone works on getting UE stuff and even the UA Totem barbarians added in the midst of all this.

**Comment by imurrx** [comment]
I would too.

**Comment by JoeyD473** [comment]
I'm a back end programmer. Haven't done anything in VBA, but may be able to
help with stand alone depending on the language

Joey

**Comment by Heliomance** [comment]
Wish Git could diff the excell file properly - then I could start a branch for adding new content while wrestling with how to make the saving work

**Comment by Heliomance** [comment]
Hrm. I wonder which would be easier - writing the VBA code to save characters as JSON, or writing a converter in a sensible language that can interpret the save file from HF as it is and output JSON.

EDIT: HF saves as an Excel sheet, not as a CSV. Probably writing the VBA then, annoyingly

**Comment by ThAlEdison** [comment]
Well, you can, with some trouble, read it using a database connection:
https://www.connectionstrings.com/excel-2003/
I've never been able to put in the time, but I know a bit about .Net,
PowerShell, C++, and SQL.

**Comment by Heliomance** [comment]
Ah-ha! I found a library that makes JSON manipulation easy. My world is no longer pain.

**Comment by Heliomance** [comment]
Could y'all go have a look at #122 and weigh in please?

**Comment by TwistedFaith84** [comment]
If it would help I have a HUGE DB of most (if not ALL) monster, Creatures for 3.5 that I found online at one point a while ago.
